src/js/common/components/RateLimiter.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import LoadingIndicator from './LoadingIndicator';
+
+/*
+ * Expects a settings object that looks like:
+ *
+ * settings: {
+ *   [id]: {
+ *     rateLimitAuthed: 1,
+ *     rateLimitUnauthed: 1
+ *   }
+ * }
+ */
+export class RateLimiter extends React.Component {
+  constructor(props) {
+    super(props);
+    // 0 is nothing gets through, 1 is everything gets through
+    const { rateLimitAuthed = 1, rateLimitUnauthed = 1 } = window.settings[props.id] || {};
+    const randomizer = Math.random();
+
+    this.state = {
+      disableRateLimit: window.sessionStorage.getItem(`${props.id}_disableRateLimit`),
+      rateLimitedUnauthed: randomizer >= rateLimitUnauthed,
+      rateLimitedAuthed: randomizer >= rateLimitAuthed
+    };
+  }
+
+  componentDidMount() {
+    this.disableRateLimitIfNecessary();
+  }
+
+  componentDidUpdate() {
+    this.disableRateLimitIfNecessary();
+  }
+
+  disableRateLimitIfNecessary = () => {
+    const { state, waitForProfile, id } = this.props;
+    const { rateLimitedUnauthed, rateLimitedAuthed } = this.state;
+
+    if ((!state.user.profile.loading || !waitForProfile) && (
+      (state.user.login.currentlyLoggedIn && !rateLimitedAuthed) ||
+      (!state.user.login.currentlyLoggedIn && !rateLimitedUnauthed)
+    )) {
+      window.sessionStorage.setItem(`${id}_disableRateLimit`, 'true');
+    }
+  }
+
+  render() {
+    const { state, bypassLimiter, renderLimitedContent, waitForProfile, children } = this.props;
+
+    if (waitForProfile && state.user.profile.loading) {
+      return <LoadingIndicator message="Loading your profile information..."/>;
+    }
+
+    const rateLimited = state.user.login.currentlyLoggedIn
```

Comment:
Should we set up rateNotLimited instead, since we only seem to check if it is not true?